File: Rotting_Oranges/Rotting-Oranges.py
from copy import deepcopy

board = [[2,1,0],[0,1,1],[1,0,1],[2,0,1]]
width = len(board)
length = len(board[0])

def check_board(static_board):
    for i in range(width):
        for j in range(length):
            if static_board[i][j] == 2:
                rot_oranges(board,i,j)

    
def rot_oranges(board,row,col):
    #check left
    if row-1 >= 0:
        if board[row-1][col] == 1:
            board[row-1][col] = 2
    #check right
    if row+1 < width:
        if board[row+1][col] == 1:
            board[row+1][col] = 2
    #check above
    if col-1 >= 0:
        if board[row][col-1] == 1:
            board[row][col-1] = 2
    
    #check below
    if col+1 < length: 
        if board[row][col+1] == 1:   
            board[row][col+1] = 2

def still_fresh(static_board):
    for i in range(width):
        for j in range(length):
            if static_board[i][j] == 1:
                return True
    return False

if __name__ == "__main__":
    minute = 0
    while(True):
        static_board = deepcopy(board)
        check_board(static_board)
        if static_board == board:
            if still_fresh(static_board):
                print "Couldnt get them all"
                minute = -1
                break
            else:
                print "All Oranges have been rotten"
                break
        minute += 1
        
        
        print board
    print minute

File: c32/Rotting-Oranges.go
package main
import "fmt"

const rows = 1
const cols = 2
var board = [rows][cols]int{  
   {0,2},
}

func rot_others(i int, j int) {
  //check right
  if i+1 < rows{
    if board[i+1][j] == 1 {
      board[i+1][j] = 2
    }
    
  }
  //check left
  if i-1 >= 0 {
    if board[i-1][j] == 1 {
      board[i-1][j] = 2
    }
  }
  //check above
  if j+1 < cols{
    if board[i][j+1] ==1 {
      board[i][j+1] = 2
    }
  }
  //check below
  if j-1 >= 0{
    if board[i][j-1] ==1 {
      board[i][j-1] = 2
    }
  }
}

func check_board(newboard [rows][cols]int){
  for i := 0; i < rows; i++ {
    for j := 0; j < cols; j++{
      if newboard[i][j] == 2{
        rot_others(i,j)
      }
    }
  }
}
func still_fresh(newboard [rows][cols]int) bool{
  for i := 0; i < rows; i++ {
    for j := 0; j < cols; j++{
      if newboard[i][j] == 1{
        return true
      }
    }
  }
  return false
}

func main() {
  for minute := 0; minute < 10; minute ++{
    var tmpboard = board
    check_board(tmpboard)
    if board == tmpboard{
      if still_fresh(tmpboard) {
        fmt.Print("couldnt get them all")
        minute = -1
        fmt.Println(minute)
        break
      } else {
        fmt.Println("All orages have been currupt")
        fmt.Println(minute)
        break
      }
    }
    fmt.Println(board)
  }
  
}